src/librbd/AioObjectRequest.h

```diff
@@ -93,6 +96,7 @@ class AioObjectRequest : public AioObjectRequestHandle {
   Context *m_completion;
   Extents m_parent_extents;
   bool m_hide_enoent;
+  const blkin_trace_info *m_trace_info;
```

Comment:
as @dillaman pointed out, it's not safe to hold on to the trace info pointers like this

take a look at how Objecter and Op work for an example; they pass optional traces around by `ZTracer::Trace*`, and store them by value as `ZTracer::Trace`. we should only be using `blkin_trace_info` in our public interface (anything in src/include/rados/), then turning it into a `Trace` object as soon as we can